pkg/acquisition/modules/http/http.go

```diff
@@ -0,0 +1,415 @@
+package httpacquisition
+
+import (
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+
+	"gopkg.in/tomb.v2"
+	"gopkg.in/yaml.v3"
+
+	"github.com/crowdsecurity/go-cs-lib/trace"
+
+	"github.com/crowdsecurity/crowdsec/pkg/acquisition/configuration"
+	"github.com/crowdsecurity/crowdsec/pkg/types"
+)
+
+var (
+	dataSourceName = "http"
+)
+
+var linesRead = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "cs_httpsource_hits_total",
+		Help: "Total lines that were read from http source",
+	},
+	[]string{"path", "src"})
+
+type HttpConfiguration struct {
+	//IPFilter                        []string          `yaml:"ip_filter"`
+	//ChunkSize                         *int64             `yaml:"chunk_size"`
+	ListenAddr                        string             `yaml:"listen_addr"`
+	Path                              string             `yaml:"path"`
+	AuthType                          string             `yaml:"auth_type"`
+	BasicAuth                         *BasicAuthConfig   `yaml:"basic_auth"`
+	Headers                           *map[string]string `yaml:"headers"`
+	TLS                               *TLSConfig         `yaml:"tls"`
+	CustomStatusCode                  *int               `yaml:"custom_status_code"`
+	CustomHeaders                     *map[string]string `yaml:"custom_headers"`
+	MaxBodySize                       *int64             `yaml:"max_body_size"`
+	Timeout                           *time.Duration     `yaml:"timeout"`
+	configuration.DataSourceCommonCfg `yaml:",inline"`
+}
+
+type BasicAuthConfig struct {
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+}
+
+type TLSConfig struct {
+	InsecureSkipVerify bool   `yaml:"insecure_skip_verify"`
+	ServerCert         string `yaml:"server_cert"`
+	ServerKey          string `yaml:"server_key"`
+	CaCert             string `yaml:"ca_cert"`
+}
+
+type HTTPSource struct {
+	metricsLevel int
+	Config       HttpConfiguration
+	logger       *log.Entry
+	Server       *http.Server
+}
+
+func (h *HTTPSource) GetUuid() string {
+	return h.Config.UniqueId
+}
+
+func (h *HTTPSource) UnmarshalConfig(yamlConfig []byte) error {
+	h.Config = HttpConfiguration{}
+	err := yaml.Unmarshal(yamlConfig, &h.Config)
+	if err != nil {
+		return fmt.Errorf("cannot parse %s datasource configuration: %w", dataSourceName, err)
+	}
+
+	if h.Config.Mode == "" {
+		h.Config.Mode = configuration.TAIL_MODE
+	}
+
+	return nil
+}
+
+func (hc *HttpConfiguration) Validate() error {
+	if hc.ListenAddr == "" {
+		return errors.New("listen_addr is required")
+	}
+
+	if hc.Path == "" {
+		hc.Path = "/"
+	}
+	if hc.Path[0] != '/' {
+		return errors.New("path must start with /")
+	}
+
+	switch hc.AuthType {
```

Comment:
The errors returned here do not provide context as to why `X is required`

I would expect it to say "Basic auth is selected but no X is provided"

This is small nit pick but could provide enough error context to user to self resolve